src/mrsimulator/spin_system/isotope.py

```diff
@@ -14,6 +14,8 @@
 MODULE_DIR = path.dirname(path.abspath(__file__))
 ISOTOPE_DATA = loadfn(path.join(MODULE_DIR, "isotope_data.json"))
 
+custom_isotope_data = {}
```

Comment:
Avoid using global variable. Make `custom_isotope_data` as a class attribute. All isotope objects will share the class attribute.